frontend/src/metabase/query_builder/components/view/ViewHeader/components/ToggleNativeQueryPreview/ToggleNativeQueryPreview.styled.tsx

```diff
@@ -3,12 +3,21 @@ import styled from "@emotion/styled";
 import IconButtonWrapper from "metabase/components/IconButtonWrapper";
 import { color } from "metabase/lib/colors";
 
-export const SqlButton = styled(IconButtonWrapper)`
-  color: ${color("text-dark")};
-  padding: 0.5rem;
+interface SqlButtonProps {
+  isSelected?: boolean;
+}
+
+export const SqlButton = styled(IconButtonWrapper)<SqlButtonProps>`
+  color: ${({ isSelected }) =>
+    isSelected ? color("white") : color("text-dark")};
+  background-color: ${({ isSelected }) => isSelected && color("brand")};
+  height: 2rem;
+  width: 2rem;
 
   &:hover {
     color: ${color("brand")};
     background-color: ${color("bg-medium")};
+    border: 1px solid ${color("brand")};
+    transition: all 200ms linear;
```

Comment:
We have way too many different ways of styling a simple icon wrapper.
I've reluctantly added the styling that mimics the notebook icon, for example, but I'll make sure to follow up with a PR that cleans all this.

Reply:
https://github.com/metabase/metabase/issues/40523